pkg/controllers/infracluster/infracluster_controller.go

```diff
@@ -227,26 +218,49 @@ func (r *InfraClusterController) ensureInfraCluster(ctx context.Context, log log
 	return infraCluster, nil
 }
 
-// setAvailableCondition sets the ClusterOperator status condition to Available.
-func (r *InfraClusterController) setAvailableCondition(ctx context.Context, log logr.Logger) error {
+// setControllerConditionsToNormal sets the InfraClusterController conditions to the normal state.
+func (r *InfraClusterController) setControllerConditionsToNormal(ctx context.Context, log logr.Logger) error {
 	co, err := r.GetOrCreateClusterOperator(ctx)
 	if err != nil {
 		return fmt.Errorf("failed to get cluster operator: %w", err)
 	}
 
 	conds := []configv1.ClusterOperatorStatusCondition{
-		operatorstatus.NewClusterOperatorStatusCondition(InfraClusterControllerAvailableCondition, configv1.ConditionTrue, operatorstatus.ReasonAsExpected,
+		operatorstatus.NewClusterOperatorStatusCondition(operatorstatus.InfraClusterControllerAvailableCondition, configv1.ConditionTrue, operatorstatus.ReasonAsExpected,
 			"InfraCluster Controller works as expected"),
-		operatorstatus.NewClusterOperatorStatusCondition(InfraClusterControllerDegradedCondition, configv1.ConditionFalse, operatorstatus.ReasonAsExpected,
+		operatorstatus.NewClusterOperatorStatusCondition(operatorstatus.InfraClusterControllerDegradedCondition, configv1.ConditionFalse, operatorstatus.ReasonAsExpected,
 			"InfraCluster Controller works as expected"),
 	}
 
-	co.Status.Versions = []configv1.OperandVersion{{Name: controllers.OperatorVersionKey, Version: r.ReleaseVersion}}
+	log.V(2).Info("InfraCluster Controller is Available")
+
+	if err := r.SyncStatus(ctx, co, conds); err != nil {
+		return fmt.Errorf("failed to sync cluster operator status: %w", err)
+	}
+
+	return nil
+}
+
+// setControllerConditionDegraded sets the InfraClusterController conditions to a degraded state.
+//
+//nolint:unused
```

Comment:
I'm reviewing commit-wise, so this might be used later in the PR, but if not, we should delete it and add it when it's actually called.

Reply:
Yeah, if this isn't used, lets dump it